cmd/pctl/add.go

```diff
@@ -245,6 +250,28 @@ func getGitRepositoryNamespaceAndName(c *cli.Context) (string, string, error) {
 	return "", "", fmt.Errorf("flux git repository not provided, please provide the --git-repository flag or use the pctl bootstrap functionality")
 }
 
+// getOutFolder returns the output folder with the following precedence:
+// User set --out overrides local configuration.
+// Local configuration, if set.
+// Default out which is `.`.
+func getOutFolder(c *cli.Context) (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch current working directory: %w", err)
+	}
+	config, err := bootstrap.GetConfig(wd)
+	if err != nil {
+		return "", err
+	}
+	if c.IsSet("out") {
+		return c.String("out"), nil
+	}
+	if config != nil && config.DefaultDir != "" {
+		return config.DefaultDir, nil
+	}
+	return defaultOut, nil
+}
+
```

Comment:
Missing some tests! A lot of logic going down here

![image](https://user-images.githubusercontent.com/19669095/133620678-ff220123-eb65-4705-99d6-1938e3463b6f.png)

Reply:
For the record, you don't have any tests for your git-repository in add either. :D

Reply:
Ah, you do, but in e2e...........

Reply:
I was about to write a whole thing in `add` for this. :D It didn't occur to me to look in e2e 🤦

Reply:
Also, that's still Eddy Murphy to me. :D

Reply:
> Ah, you do, but in e2e...........

:smirk: 

but for real yeah unit testing this file SUCKS. So a lot of the tests end up integration, which SUCKS even more. Add unit tests if you can :smile:

Reply:
Also `do as i say not as i do` applies here :joy: